In [1]:
import pandas as pd
import numpy as np

In [2]:
df_iris = pd.read_csv("https://raw.githubusercontent.com/plotly/datasets/master/iris-data.csv")

In [3]:
# Muestra el tamaño del dataset iris
df_iris.shape

(150, 5)

In [4]:
#Muestra las primeras cinco filas y permite además ver las cinco columnas que componen el dataset iris
df_iris.head()

sepal length  sepal width  petal length  petal width        class
0           5.1          3.5           1.4          0.2  Iris-setosa
1           4.9          3.0           1.4          0.2  Iris-setosa
2           4.7          3.2           1.3          0.2  Iris-setosa
3           4.6          3.1           1.5          0.2  Iris-setosa
4           5.0          3.6           1.4          0.2  Iris-setosa

In [5]:
df_iris.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   sepal length  150 non-null    float64
 1   sepal width   150 non-null    float64
 2   petal length  150 non-null    float64
 3   petal width   150 non-null    float64
 4   class         150 non-null    object 
dtypes: float64(4), object(1)
memory usage: 6.0+ KB


El dataset iris tiene cinco columnas, las cuales son: **sepal length**, que representa el largo del sépalo de las plantas del género *Iris*; **sepal width**, que representa el ancho del sépalo de las plantas anteriormente mencionadas; **petal length**, que representa el largo de los pétalos de las flores *Iris*; **petal width**, que representa el ancho de los pétalos de las plantas *Iris*; y **class** que representa a las tres especies del género *Iris* (*Iris setosa*, *Iris versicolor* e *Iris virginica*). Las columnas numéricas contienen datos del tipo float, mientras que la columna **class** contiene datos del tipo string, el cual es considerado como un dato de tipo objeto en el dataframe. 

In [6]:
df_iris.isna().sum()

sepal length    0
sepal width     0
petal length    0
petal width     0
class           0
dtype: int64

No existen **valores nulos** en las cinco columnas del dataset Iris.

In [7]:
df_iris.describe()

sepal length  sepal width  petal length  petal width
count    150.000000   150.000000    150.000000   150.000000
mean       5.843333     3.054000      3.758667     1.198667
std        0.828066     0.433594      1.764420     0.763161
min        4.300000     2.000000      1.000000     0.100000
25%        5.100000     2.800000      1.600000     0.300000
50%        5.800000     3.000000      4.350000     1.300000
75%        6.400000     3.300000      5.100000     1.800000
max        7.900000     4.400000      6.900000     2.500000

La columna **sepal length** contiene 150 valores numéricos del tipo float. El **valor promedio** del largo del sépalo es igual a 5,84 cm. La **desviación estándar** de los valores del largo de sépalo es igual a 0,83 cm. El **valor mínimo** del largo de sépalo es igual a 4,3 cm, mientras que el **valor máximo** del largo de sépalo es igual a 7,9 cm, por lo tanto, su **rango** es igual a 3,6 cm. El valor del **percentil 25** es igual a 5,1 cm, mientras que el valor del **percentil 50** o **mediana** es igual a 5,8 cm, y el valor del **percentil 75** es igual a 6,4 cm.

La columna **sepal width** contiene 150 valores numéricos del tipo float. El **valor promedio** del ancho del sépalo es igual a 3,05 cm. La **desviación estándar** de los valores del ancho de sépalo es igual a 0,43 cm. El **valor mínimo** del ancho de sépalo es igual a 2,0 cm, mientras que el **valor máximo** del ancho de sépalo es igual a 4,4 cm, por lo tanto, su **rango** es igual a 2,4 cm. El valor del **percentil 25** es igual a 2,8 cm, mientras que el valor del **percentil 50** o **mediana** es igual a 3,0 cm, y el valor del **percentil 75** es igual a 3,3 cm.

La columna **petal length** contiene 150 valores numéricos del tipo float. El **valor promedio** del largo de los pétalos es igual a 3,76 cm. La **desviación estándar** de los valores del largo de los pétalos es igual a 1,76 cm. El **valor mínimo** del largo de los pétalos es igual a 1,0 cm, mientras que el **valor máximo** del largo de los pétalos es igual a 6,9 cm, por lo tanto, su **rango** es igual a 5,9 cm. El valor del **percentil 25** es igual a 1,6 cm, mientras que el valor del **percentil 50** o **mediana** es igual a 4,4 cm, y el valor del **percentil 75** es igual a 5,1 cm.

La columna **petal width** contiene 150 valores numéricos del tipo float. El **valor promedio** del ancho de los pétalos es igual a 1,20 cm. La **desviación estándar** de los valores del ancho de los pétalos es igual a 0,76 cm. El **valor mínimo** del ancho de los pétalos es igual a 0,1 cm, mientras que el **valor máximo** del ancho de los pétalos es igual a 2,5 cm, por lo tanto, su **rango** es igual a 2,4 cm. El valor del **percentil 25** es igual a 0,3 cm, mientras que el valor del **percentil 50** o **mediana** es igual a 1,3 cm, y el valor del **percentil 75** es igual a 1,8 cm.


In [8]:
df_iris["class"].value_counts()

class
Iris-setosa        50
Iris-versicolor    50
Iris-virginica     50
Name: count, dtype: int64

Existen 50 filas para la flor *Iris-setosa*, 50 filas para la flor *Iris-versicolor* y 50 filas para la flor *Iris-virginica*.